ignite/handlers/param_scheduler.py

```diff
@@ -313,6 +315,18 @@ def __init__(
         self.cycle = 0
         self.start_value_mult = start_value_mult
         self.end_value_mult = end_value_mult
+        self.warmup_each_cycle = warmup_each_cycle
+        if not self.warmup_each_cycle:
+            if warmup_duration is not None:
+                warnings.warn(
+                    f"warmup_each_cycle=False but your warmup_duration is {warmup_duration}. "
+                    "so warmup_duration will be set to 0. "
+                    "If you want to use warmup each cycle, pleas set warmup_each_cycle=True"
+                )
+            self.warmup_duration = 0
+        else:
+            self.warmup_duration = warmup_duration
```

Comment:
What if `warmup_duration` is `None`? In the next line we add it to `self.cycle_size`

Reply:
> What if `warmup_duration` is `None`? In the next line we add it to `self.cycle_size`

oh, I didn't care about that. I will add code that handle when warmup_duration is none!